src/components/Editor/Footer.js

```diff
@@ -41,12 +41,19 @@ type Props = {
 
 class SourceFooter extends PureComponent<Props> {
   prettyPrintButton() {
-    const { selectedSource, togglePrettyPrint } = this.props;
+    const { selectedSource, togglePrettyPrint, isPrettyPrinting } = this.props;
     const sourceLoaded = selectedSource && isLoaded(selectedSource);
 
     if (!shouldShowPrettyPrint(selectedSource)) {
       return;
     }
+    if (isLoading(selectedSource) || isPrettyPrinting) {
```

Comment:
do we need `isPrettyPrinting` or could we use `selectedSource.prettyPrinted`

Reply:
When we pretty print a source, we generate a new pretty printed source, and then select it, we don't update the current source ands set prettyPrinted to be true.

Reply:
so if we pretty print and then immediately select it we should be good. I think that will help to because the editor will show loading immediately as well